migrations/0029_index_queries.py

```diff
@@ -0,0 +1,28 @@
+from playhouse.migrate import PostgresqlMigrator, migrate
+
+from redash.models import db
+from redash import models
+
+if __name__ == '__main__':
+    db.connect_db()
+    migrator = PostgresqlMigrator(db.database)
+
+    with db.database.transaction():
+        migrate(
+            migrator.add_column('queries', 'tsv', models.Query.tsv)
+        )
+        db.database.execute_sql("CREATE INDEX tsv_idx ON queries USING gin(tsv)")
+        db.database.execute_sql("UPDATE queries SET tsv = setweight(to_tsvector(coalesce(name, '')), 'A') || setweight(to_tsvector(coalesce(description, '')), 'B') || setweight(to_tsvector(coalesce(query, '')), 'C')")
+        db.database.execute_sql("""
+CREATE FUNCTION query_search_trigger() RETURNS trigger AS $f$
```

Comment:
The migration will work for existing deployments. But what about new ones? Have you considered an before_save/after_save hook instead of a DB trigger?

Also, did you check if it's too much of a performance hit if we do the calculation of the vector in "realtime" instead of storing it?